### DISCUSSION TITLE
LEDのはんだ付けとテスターについて

### DISCUSSION THREAD

**Discussion by ntnl-it-wiz** [Yowkees/keyball#336] [Q&A]
LEDですが、他のディスカッションと以下の仕様書にてDIN、DOUT、VDD、GNDの4つの足であることがわかりました。
http://yushakobo.jp/ds/YS-SK6812MINI-E.pdf

うまくつなげていない場合（点灯していない場合）どのように確認すればよいかご教示いただけないでしょうか。

Keyballの作成マニュアルには以下の記述がありました。

> LEDは直列に配線されているので、途中までしか光らない場合は修正が必要です。
> 例えば12番のLEDまでは点灯したが、13番以降のLED以降が点灯しない場合、【12番か13番】のLEDの半田付けに問題がある可能性が高いのでそれらのLEDの半田付けをやり直してください。それでも改善しない場合はLEDの故障が考えられるので13番のLEDを交換してください。

該当内容より、もし13番が点灯しない場合、おおよそ以下の原因があると想定しました。
- A. 12番のDOUT、13番のDIN接触不良
- B. 13番のGND、VDDの接触不良
- C. 13番のLEDの個体不良

Aのケースでは12番のDOUTと13番のDINのはんだ部分を導通確認すればはんだ付けがうまくいっているか確認できるのかと想定しております。
Bのケースは、13番といずれかのLEDのGND同士、VDD同士と導通確認すれば、はんだ付けができていると確認できるのではと想定しております。
Cのケースは、LEDのGNDから各端子と導通確認をして、適当な値が出てくれば内部回路の抵抗値を示していると考えて利用可能という想定でおります。

以上の前提でテスターによる確認をした限りでは、いずれも接続できているのに13番が点灯しないという状況になり、どのように調査・修正するべきか詰まっております。
上記の想定に関しての認識齟齬や他の確認項目がありましたら指摘いただけますと幸いです。

なお、一旦LEDはおいておきキーボード、トラックボール自体は完成させて動作確認しております。
一部しかLEDが光らないことを除けば正常に使えております。
また、他のディスカッションで挙がっていたエナメル線はAmazon配送待ちでありまだ試せておりません。

**Comment by ntnl-it-wiz** [comment]
本日エナメル線が届きましたので切り分けのバイパス処理として12番 DOUT - 14番 DINの接続をしたところ14番は点灯いたしました。13番が個体不良かを単体で確認する方法や13番が点灯しない被疑箇所についてご教示いただけますと幸いです。

**Comment by Yowkees** [comment]
ご連絡が遅くなり申し訳ありませんでした。

文章を読ませていただいた印象としては、導通確認等問題なかったということですので、やはり13番LEDの故障の可能性が高いと思います。
思い切って交換してしまうことをお勧めします。
なお、LEDの除去は半田こてを２本使用すると、とても簡単に外せますので、安価な半田こてで結構ですので、２本使用することをお勧めします。

お手数ですがご確認お願いいたします。

**Reply by ntnl-it-wiz** [reply]
新しいはんだごてが届いたので取り外して差し替えてみました。
差し替え後は無事点灯いたしました。

まだ全部終わっておりませんが、上の確認方法でダメだったら差し替えという方式で引き続き完成目指して頑張ります。
回答いただきありがとうございました。
